ampd/src/block_height_monitor.rs

```diff
@@ -0,0 +1,148 @@
+use error_stack::{Result, ResultExt};
+use std::{
+    sync::{atomic::AtomicU64, Arc},
+    time::Duration,
+};
+use thiserror::Error;
+use tokio::{select, time};
+use tokio_util::sync::CancellationToken;
+use tracing::info;
+
+use crate::tm_client::TmClient;
+
+#[derive(Clone)]
```

Comment:
```suggestion
```
It doesn't make sense to me to clone the entire monitor so that other threads can read. It doesn't make sense for them to be able to `.run()` again.